tests/test_transforms.py

```diff
@@ -5,24 +5,31 @@
 
 params_q_to_tth = [
     # UC1: Empty q values, no wavelength, return empty arrays
-    ([None, np.empty((0))], np.empty((0))),
-    # UC2: Empty q values, wavelength specified, return empty arrays
-    ([4 * np.pi, np.empty((0))], np.empty(0)),
+    (None, np.empty((0)), np.empty((0))),
+    # # UC2: Empty q values, wavelength specified, return empty arrays
+    (4 * np.pi, np.empty((0)), np.empty(0)),
     # UC3: User specified valid q values, no wavelength, return empty arrays
     (
-        [None, np.array([0, 0.2, 0.4, 0.6, 0.8, 1])],
+        None,
+        np.array([0, 0.2, 0.4, 0.6, 0.8, 1]),
         np.array([0, 1, 2, 3, 4, 5]),
     ),
     # UC4: User specified valid q values (with wavelength)
     # expected tth values are 2*arcsin(q) in degrees
-    ([4 * np.pi, np.array([0, 1 / np.sqrt(2), 1.0])], np.array([0, 90.0, 180.0])),
+    (4 * np.pi, np.array([0, 1 / np.sqrt(2), 1.0]), np.array([0, 90.0, 180.0])),
 ]
 
 
-@pytest.mark.parametrize("inputs, expected", params_q_to_tth)
-def test_q_to_tth(inputs, expected):
-    actual = q_to_tth(inputs[1], inputs[0])
-    assert np.allclose(expected, actual)
+@pytest.mark.parametrize("wavelength, q, expected_tth", params_q_to_tth)
```

Comment:
@sbillinge @yucongalicechen 

Before addressing other pytest warnings, I just discuss our group's way for writing tests. 

Here, instead of passing `input[1]` or `input[0]`, it's actually difficult for me (new user) to understand what these variables refer to. 

Moving forward, wouldn't it easier to specify them as `wavelength` and `q` next to  `mark.parametrize` instead? so that we can directly use these methods and the tests are much. more readable.

Reply:
yes, this makes sense.  I agree.   Since tests test behavior, the easier it is to see the behvaior, the better.  Sometimes it can be a bit hard with the `parameterize` decorator, but this is a good way to make it easier.  I agree.